packaging/flavorgen/cmd/root.go

```diff
@@ -52,30 +79,74 @@ func RunRoot(command *cobra.Command) error {
 	if err != nil {
 		return errors.Wrapf(err, "error accessing flag %s for command %s", flavorFlag, command.Name())
 	}
+	outputDir, err := command.Flags().GetString(outputDirFlag)
+	if err != nil {
+		return errors.Wrapf(err, "error accessing flag %s for command %s", outputDirFlag, command.Name())
+	}
+	var outputFlavors []string
+	if flavor != "" {
+		outputFlavors = append(outputFlavors, flavor)
+	} else {
+		outputFlavors = allFlavors
+	}
+	generateMultiFlavors := len(outputFlavors) > 1
+	for _, f := range outputFlavors {
+		manifest, err := generateSingle(f)
+		if err != nil {
+			return err
+		}
+
+		fileMapping, ok := flavorMappings[f]
+		if !ok {
+			return fmt.Errorf("file mapping for flavor %q is missng in flavorMappings", f)
+		}
+
+		yamlFileName := fileMapping + ".yaml"
+		if outputDir == "-" {
+			if generateMultiFlavors {
+				// use the yaml filename as a section delimiter
+				fmt.Printf("### %s\n", yamlFileName)
+			}
+			fmt.Print(manifest)
+			continue
+		}
+
+		yamlPath := filepath.Join(outputDir, yamlFileName)
+		err = os.WriteFile(yamlPath, []byte(manifest), 0600)
+		if err != nil {
+			return errors.Wrapf(err, "failed to dump manifest content to file for flavor %s", f)
+		}
+	}
+
+	return nil
+}
+
+func generateSingle(flavor string) (string, error) {
+	util.Replacements = append([]util.Replacement{}, util.DefaultReplacements...)
```

Comment:
Really the last improvement:

```suggestion
	replacements := util.DefaultReplacements
```

Let's make this private and make it a parameter for `util.GenerateManifestYaml(objs, replacements)` :-)

Reply:
This idea was considered in my last commit.
It was dropped as there are a few dependencies on the global variable `util.Replacements`.
I think it would be better to use a separate PR to move it as parameters instead of global variable.

Reply:
Hm, I thought it is only:
* the `root.go` which needs to change it for the one case 
* the `GenerateObjectYAML` function itself which reads it (and instead we could pass it to this func as parameter)
* https://github.com/chrischdi/cluster-api-provider-vsphere/blob/671aea8af0aa3a368361dae6bc1330f1b0a2d8f8/packaging/flavorgen/flavors/crs/util.go#L81 which also calls `GenerateObjectYAML`, there we could simply pass `util.DefaultReplacements` as parameter.

Reply:
But I'm also ok with the code as is, this is more cosmetical and not a functional idea/change.

Reply:
Yep. Those are the few lines I'm not 100% sure.
The generated files are the same and guessing it should be okay to adopt these changes as well?
I will push the change and let me know if you have other concerns. :)